src/Symfony/Component/AssetMapper/Factory/MappedAssetFactory.php

```diff
@@ -25,13 +25,16 @@ class MappedAssetFactory implements MappedAssetFactoryInterface
 {
     private const PREDIGESTED_REGEX = '/-([0-9a-zA-Z]{7,128}\.digested)/';
 
+    private const DEFAULT_HASH_ALGORITHM = 'xxh128';
+
     private array $assetsCache = [];
     private array $assetsBeingCreated = [];
 
     public function __construct(
         private readonly PublicAssetsPathResolverInterface $assetsPathResolver,
         private readonly AssetMapperCompiler $compiler,
         private readonly string $vendorDir,
+        private readonly string $hashAlgorithm = self::DEFAULT_HASH_ALGORITHM,
```

Comment:
Should the argument be validated against `hash_algos()`?

Reply:
Fixed, thanks!